### PR TITLE
Linux 4.12 compat: super_setup_bdi_name()

### PR DESCRIPTION
**Pull request by behlendorf**
All filesystems were converted to dynamically allocated BDIs.  The
destruction of backing_dev_info structures is handled as part of
super block destruction.

### CODE REVIEW

File: config/kernel-bdi.m4 [hunk 1]
```diff
@@ -0,0 +1,56 @@
+dnl #
+dnl # 2.6.32 - 2.6.33, bdi_setup_and_register() is not exported.
+dnl # 2.6.34 - 3.19, bdi_setup_and_register() takes 3 arguments.
+dnl # 4.0 - 4.10, bdi_setup_and_register() takes 2 arguments.
+dnl # 4.11 - x.y, super_setup_bdi_name() new interface.
```
Comment (tuxoko) [inline config/kernel-bdi.m4]: 4.12?

File: module/zfs/zfs_vfsops.c [hunk 2]
```diff
@@ -1606,9 +1607,8 @@ zfs_domount(struct super_block *sb, zfs_mnt_t *zm, int silent)
 	sb->s_blocksize = recordsize;
 	sb->s_blocksize_bits = ilog2(recordsize);
 	zfsvfs->z_bdi.ra_pages = 0;
```
Comment (tuxoko) [inline module/zfs/zfs_vfsops.c]: Do we need to modify sb->s_bdi->ra_pages for 4.12+?

Also, we may want to get rid of z_bdi to save space.

**Comment by behlendorf** [comment]
@tuxoko good points, how about this.  The updated patch refactors the code such that all backing_dev_info struct management is handled in the wrapper functions.  After a successful call to `zpl_bdi_setup()` the `sb->s_sb` will contain a pointer to the backing_dev_info, it should be destroyed by `zpl_bdi_destroy()`.  This allowed us to remove the `z_bsi` from the `zfsvfs_t` and rely on the pointer in the super block.

**Comment by tuxoko** [comment]
One nit, the commit title should be 4.12

**Comment by behlendorf** [comment]
Indeed, I'll fix it in the merge unless I need to refresh this for other reasons.  Thanks.

**Pull request merged**